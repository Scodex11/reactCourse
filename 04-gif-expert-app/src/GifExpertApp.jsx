import { useState } from 'react'
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const arrayCategories = ['Dragon Ball']
    const [categories, setCategories] = useState(arrayCategories)

    
    const onNewCategory = (newCategory) => {
        
        if (categories.includes(newCategory)){
            alert("¡La categoría ya existe!");
            return;
        } //No se permite agregar categorias repetidas

        setCategories([newCategory, ...categories])
        console.log(categories)
    }


    return (
    <>
        <h1>GifExpertApp</h1>

        {/* INPUT */}
        <AddCategory onNewCategory = { onNewCategory } />    
    
       
        {
            categories.map( category => (
                <GifGrid 
                    key = { category }
                    category = { category }
                />
            ))
        }


    </>
  )
}
