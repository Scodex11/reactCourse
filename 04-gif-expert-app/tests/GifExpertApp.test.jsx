import {fireEvent, screen, render} from '@testing-library/react';
import {GifExpertApp} from '../src/GifExpertApp.jsx';

describe('Pruebas en GifExpertApp', () => { 
    
    test('Debe de mostrar el componente correctamente', () => {
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    })

    test('Debe de agregar una categoria nueva', () => {
        const categories = ['Droga en bolsita'];

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: categories[0]}});        
        fireEvent.submit(form);

        /*Se imprime por pantalla la nueva categoria */

    })


 })