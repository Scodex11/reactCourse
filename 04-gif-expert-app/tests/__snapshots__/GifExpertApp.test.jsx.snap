// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Escribe la categoria del gif..."
      type="text"
      value=""
    />
  </form>
  <h3>
    Dragon Ball
  </h3>
  <p>
    Cargando...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
